tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+	"compilerOptions": {
+		"module": "commonjs",
+		"declaration": true,
+		"noImplicitAny": true,
+		"removeComments": true,
+		"preserveConstEnums": true,
+		"strictNullChecks": true,
+		"sourceMap": true,
+		"target": "es5",
+		"noUnusedParameters": true,
+		"noUnusedLocals": true
```

Comment:
Imo, as a general rule it would be nice to keep TS build config all in the tsconfig, as much as possible, instead of adding extra parts of it only in gulp. Obviously for cases where we can't then gulp is fine, but for the output directory at least we certainly can.

Every tool that understands typescript understands tsconfig, but not all of them will happily delegate the build process to gulp. E.g. [wallaby](https://wallabyjs.com/), many other non-VS code editors. If everything required is in tsconfig, we shouldn't need to do any special setup for other editors at all - typescript support is the only prerequisite. Plus having one canonical place for TS build config is convenient for finding and managing it.